### PR TITLE
Use submodule for libdds build

### PR DESCRIPTION
**Pull request by suokko**
Moving libdds to submodule allows front end to update in lockstep if
there will be libdds changes that is best done as an atomic update with
service changes.

Signed-off-by: Pauli <suokkos@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,6 +69,12 @@ Deploy it on the server:
 $ make deploy
 ```
 
+## Updating libdds Submodule
+
+Submodule updates can be simple done by checking out any commit hash which means
+branches, tags and plane commit hash numbers. Next step is `git add libdds`
+which stages the submodule update like other changes.
+
```
Comment (tameware) [inline README.md:77]: I have no idea what this means. Please help me out! Also. this could frighten potential newcomers. It might be better to document it otherwise.

Comment (tameware) [reply README.md:77]: I was able to deploy successfully using this branch, so all looks good once the doc is updated.

Comment (suokko) [reply README.md]: > I have no idea what this means. Please help me out! Also. this could frighten potential newcomers. It might be better to document it elsewhere.

Good point. I moved the documentation to CONTRIBUTING.md and tried to make it easier to understand.
Did I manage to write it in a manner which can be understood?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Coding Style
+
+# Issues
+
+# Pull Request
+
+## Managing Submodule
+
+If your changes require libdds changes to be include same then there is a few
```
Comment (tameware) [inline CONTRIBUTING.md:9]: "If your changes require libdds changes to be included then there are a few"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+extra steps to follow. You have to make sure required libdds changes have been
+merged to [http:://github.com/online-bridge-hackathon/libdds](http:://github.com/online-bridge-hackathon/libdds).
+
+Merging commits to libdds doesn't automatically include merges to the library
+used by service build process. The library version used by service is based on
+commit attached to the submodule state. To update the commit you need to first
```
Comment (tameware) [inline CONTRIBUTING.md:15]: "commits attached to the submodule."

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+checkout the new commit into your submodule working tree. Then the change needs
+to be staged and committed like source code changes. Following example shows
```
Comment (tameware) [inline CONTRIBUTING.md:17]: "The following example shows the"

Comment (suokko) [reply CONTRIBUTING.md]: I feel like I can't say the example sentence in plain English. It seems like my thinking process is influenced by Finnish sentence structure which doesn't map 1:1 with English.

The latest attempt: The following example shows steps to update the submodule to the latest commit in the master branch.

I'm left with feeling there would be a simpler way to say same in English.

Comment (tameware) [reply CONTRIBUTING.md]: I know how that goes. I often have the same problem, and English is my native language!

I think your latest version is fine. Slightly better might be:

"The following example shows the steps to update the submodule to the latest commit in the master branch."
